Backport af046c8 from upstream

commit af046c8987ec3c580eff8a0c7c438762377fdbca
Author: Ganael Laplanche <ganael.laplanche@martymac.org>
Date:   Wed Nov 29 20:54:12 2023 +0100

    Fix build on FreeBSD
    
    FreeBSD's alloca(3) only needs <stdlib.h>

diff --git a/src/common/byte_stream.cpp b/src/common/byte_stream.cpp
index bb26af0f..18609996 100644
--- src/common/byte_stream.cpp
+++ src/common/byte_stream.cpp
@@ -28,8 +28,10 @@
 #ifdef _MSC_VER
 #include <malloc.h>
 #else
+#if !defined(__FreeBSD__)
 #include <alloca.h>
 #endif
+#endif
 
 Log_SetChannel(ByteStream);
 
